plugins/services/src/js/components/modals/NewCreateServiceModalForm.js

```diff
@@ -240,9 +259,119 @@ class NewCreateServiceModalForm extends Component {
     return rootErrors;
   }
 
+  getContainerList(data) {
+    if (data.containers && data.containers.length !== 0) {
+      return data.containers.map((item, index) => {
+        return (
+            <TabButton key={index} id={`container${index}`}
+                label={item.name || `container ${index + 1}`}/>
+        );
+      });
+    }
+
+    return null;
+  }
+
+  getContainerContent(data, errors) {
+    let {containers = []} = data;
+    return containers.map((item, index) => {
+      let itemErrors = errors && errors.containers && errors.containers[item];
+
+      return (
+          <TabView key={index} id={`container${index}`}>
+            <PodContainerServiceFormSection
+                path={`containers.${index}`}
+                data={data}
+                errors={itemErrors}
+                onRemoveItem={this.handleRemoveItem}
+                onAddItem={this.handleAddItem} />
+          </TabView>
+      );
+    });
+  }
+
+  getSectionList() {
+    if (this.state.isPod) {
+      return [
+        <TabButton id="networking" label="Networking" key="networking" />,
+        <TabButton id="environment" label="Environment" key="environment"/>,
+        <TabButton id="healthChecks" label="Health Checks" key="healthChecks"/>
+      ];
+    }
+    return [
+      <TabButton id="networking" label="Networking" key="networking" />,
+      <TabButton id="environment" label="Environment" key="environment"/>,
+      <TabButton id="healthChecks" label="Health Checks" key="healthChecks"/>,
+      <TabButton id="volumes" label="Volumes" key="volumes"/>
+    ];
+  }
+
+  getSectionContent(data, errorMap) {
+    let rootErrorComponent = this.getRootErrorMessage();
+
+    if (this.state.isPod) {
+      return [
+        <TabView id="environment" key="environment">
+          {rootErrorComponent}
+          <EnvironmentFormSection
+              data={data}
+              errors={errorMap}
+              onRemoveItem={this.handleRemoveItem}
+              onAddItem={this.handleAddItem} />
+        </TabView>,
+        <TabView id="healthChecks" key="healthChecks">
+          <HealthChecksFormSection
+              data={data}
+              errors={errorMap}
+              onRemoveItem={this.handleRemoveItem}
+              onAddItem={this.handleAddItem} />
+        </TabView>,
+        <TabView id="networking" key="networking">
+          <NetworkingFormSection
+              data={data}
+              errors={errorMap}
+              onRemoveItem={this.handleRemoveItem}
+              onAddItem={this.handleAddItem} />
+        </TabView>
+      ];
+    }
+
+    return [
+      <TabView id="environment" key="environment">
```

Comment:
There's a lot of double declaration here.
You could have a sections array and then filter it:
```js
if (this.state.isPod) {
  return sections.filter((item) => {
    return !['volumes'].inlcudes(item.key);
  });
}

return sections;
```